packages/leancode_cubit_utils/example/lib/pages/paginated/api.dart

```diff
@@ -0,0 +1,187 @@
+import 'dart:convert';
+
+import 'package:example/http/status_codes.dart';
+import 'package:http/http.dart' as http;
+import 'package:equatable/equatable.dart';
+import 'package:faker/faker.dart';
+
+typedef Json = Map<String, dynamic>;
+
+/// This is a fake API class as an example of paginated response from an API.
+class Page {
+  Page({required this.hasNextPage, required this.users});
+
+  factory Page.fromJson(Json json) {
+    return Page(
+      hasNextPage: json['hasNextPage'] as bool,
+      users: [
+        for (final user in json['users'] as Iterable)
+          User.fromJson(user as Json),
+      ],
+    );
+  }
+
+  Json toJson() {
+    return {
+      'hasNextPage': hasNextPage,
+      'users': [
+        for (final user in users) user.toJson(),
+      ],
+    };
+  }
+
+  final bool hasNextPage;
+  final List<User> users;
+}
+
+class User {
+  User({
+    required this.name,
+    required this.jobTitle,
+  });
+
+  factory User.fake(Faker faker, String jobTitle) {
```

Comment:
The expression body is a little cleaner and easier to read. You can use it more often

Reply:
SUre, as I've said it was mostly `copy` & `paste` .